server/src/common/util/validator.js

```diff
@@ -0,0 +1,11 @@
+import { plainToClass } from 'class-transformer';
+import { validate } from 'class-validator';
+import { ValidationError } from '../error/ValidationError';
+
+export const validator = async (DtoClass, Object, groups) => {
+    const classObject = plainToClass(DtoClass, Object);
+    const [error] = await validate(classObject, { groups });
+    if (error) {
+        throw new ValidationError('validation failed');
+    }
```

Comment:
```suggestion
    await validateOrReject(classObject, { groups });
```
이렇게 하면 에러를 따로 던져주지 않아도 될 것 같아요✌️
ValidationError는 이미 'class-validator'에 정의가 되어 있어서 미들웨어인 errorHandler에서 처리하고 있습니다!
https://github.com/boostcamp-2020/Project04-B-Whale/blob/2a890211ab822d47472be513469b7538ad71483e/server/src/common/middleware/errorHandler.js#L13-L22

Reply:
validator함수가 미들웨어로 처리되는게 아니여서 그런지 validateOrReject를 사용하면 에러 핸들링이 되지 않는다는 오류가 뜨더라구요..
그래서 validateOrReject에 의해 생성되는 에러를 핸들링 하려면 코드를 아래와 같이 작성해야하는데 try-catch에서 아무런 작업없이 에러를 던지기만하는 것은 린트 설정으로 막혀져있더라구요.
``` javascript
try {
    // ... 생략
     await validateOrReject(classObject, { groups });
} catch (e) {
    throw e;
}
```
차라리 린트 설정을 수정하는게 나았을까요?🤔

Reply:
흠... 왜 처리가 안될까요
errorHandler 부분 디버깅 해주실 수 있나요?

Reply:
다시 해봤는데 아무런 문제가 없네요..
코드 작성하면서 이것저것 같이 테스트했었는데 거기서 문제가 있었었나봅니다..😂
commit 90e7527 에 리뷰 반영했습니다!